src/bootstrap/install.rs

```diff
@@ -213,6 +216,10 @@ install!((self, builder, _config),
             );
         }
     };
+    RustAnalyzer, "rust-analyzer", Self::should_build(_config), only_hosts: true, {
+        builder.ensure(dist::RustAnalyzer { compiler: self.compiler, target: self.target });
+        install_rust_analyzer(builder, self.compiler.stage, self.target);
```

Comment:
Wait, I am looking at the thing bellow, with `if should_install` and at the thing above, with `if ensure() || should_install`, and it seems like at least one of them is wrong? I think `||` should be `&&` for rls, right?

Reply:
Hm seems likely :) x.py install could definitely use some love, it's basically unmaintained and not really understood by anyone what it should do (and what it does is entirely unknown to me at least)